packages/modal/src/ux-modal-service.ts

```diff
@@ -0,0 +1,403 @@
+      activeLayer.dismiss();
+    } else if (key === 'Enter' && (keyboard === key || (Array.isArray(keyboard) && keyboard.indexOf(key) > -1))) {
+      activeLayer.ok();
+    }
+  }
+
+  private getActionKey(event: KeyboardEvent): UxModalKeybord | undefined {
+    if ((event.code || event.key) === 'Escape' || event.keyCode === 27) {
+      return 'Escape';
+    }
+    if ((event.code || event.key) === 'Enter' || event.keyCode === 13) {
+      return 'Enter';
+    }
+    return undefined;
+  }
+
+  public getLastLayer(): UxModalLayer | null {
+    return this.modalLayers.length > 0 ? this.modalLayers[this.modalLayers.length - 1] : null;
+  }
+
+  public getLastModal(): UxModal | null {
+    const lastLayer = this.getLastLayer();
+    return lastLayer !== null ? lastLayer.modal : null;
+  }
+
+  public get zIndex() {
+    return this.startingZIndex + this.modalLayers.length;
+  }
+
+  private createModalElement(options: UxModalServiceOptions, bindingContext: UxModalBindingContext): HTMLElement {
+    const element = document.createElement('ux-modal');
+    element.setAttribute('dismiss.trigger', 'dismiss()');
+    element.setAttribute('ok.trigger', 'ok($event)');
+    if (options.position !== undefined) {
+      element.setAttribute('position', options.position);
+    }
+    if (options.overlayDismiss === false) {
+      element.setAttribute('overlay-dismiss.bind', 'false');
+    }
+    if (options.outsideDismiss === false) {
+      element.setAttribute('outside-dismiss.bind', 'false');
+    }
+    if (options.lock === false) {
+      element.setAttribute('lock.bind', 'false');
+    }
+    if (options.keyboard !== undefined) {
+      bindingContext.keyboard = options.keyboard;
+      element.setAttribute('keyboard.bind', 'keyboard');
+    }
+    if (typeof options.restoreFocus === 'function') {
+      bindingContext.restoreFocus = options.restoreFocus;
+      element.setAttribute('restore-focus.bind', 'restoreFocus');
+    }
+    if (typeof options.openingCallback === 'function') {
+      bindingContext.openingCallback = options.openingCallback;
+      element.setAttribute('opening-callback.bind', 'openingCallback');
+    }
+    if (typeof options.modalBreakpoint === 'number') {
+      element.setAttribute('modal-breakpoint.bind', `${options.modalBreakpoint}`);
+    }
+
+    element.setAttribute('host.bind', 'false');
+
+    if (options.theme) {
+      bindingContext.theme = options.theme;
+      element.setAttribute('theme.bind', `theme`);
+    }
+
+    return element;
+  }
+
+  private createCompositionContext(
+    container: any, // there is a TS error if this is not any ?
+    host: Element,
+    bindingContext: UxModalBindingContext,
+    settings: {model?: any, view?: any, viewModel?: any},
+    slot?: ViewSlot
+    ): CompositionContext {
+    return {
+      container,
+      bindingContext: settings.viewModel ? null : bindingContext,
+      viewResources: this.viewResources as any, // there is a TS error if this is not any ?
```

Comment:
Hmm, maybe let's solve this later. It's ok for now I guess? Did you experience any error for not having this?

Reply:
No problem found with this `any`.